### PR TITLE
Add section regarding Theming

### PR DESCRIPTION
**Pull request by 3lvis**

### CODE REVIEW

File: BEST_PRACTICES.md [hunk 1]
```diff
@@ -195,3 +196,38 @@ var photo: Photo? {
...
+Use a `Theme.swift` that contains extensions for theming your app. When adding color extensions avoid adding the `Color` postfix to your color methods.
+
+For example:
+
+```swift
+import Hex
+import UIKit
+
+extension UIColor {
+    class func brightPurple() -> UIColor {
```
Comment (elland) [inline BEST_PRACTICES.md]: I'd go a step further and try to enforce semantic naming.

`navigationBarTintColor()` instead of `brightPurple()`.

Comment (3lvis) [inline BEST_PRACTICES.md]: That makes it tedious though, for setting a color in a different place you have to create a new function. It's double the work.

Comment (elland) [inline BEST_PRACTICES.md]: Depends on how we do it. If we have schemes like [this](https://cloud.githubusercontent.com/assets/54423/14979824/0221c706-1125-11e6-8aaf-c1b2a8e40050.png), it works fine, no?

Comment (3lvis) [inline BEST_PRACTICES.md]: Yes! That could totally work.

```
callToActionNormal
callToActionSelected
callToActionHighlighted
```

We just need to be careful to not have one method per UI element.

Comment (elland) [inline BEST_PRACTICES.md]: A worthy extra note to add. 👍

Comment (Elland) [inline BEST_PRACTICES.md]: 👍

Comment (3lvis) [inline BEST_PRACTICES.md]: Added 💥

File: BEST_PRACTICES.md [hunk 2]
```diff
@@ -199,7 +199,7 @@ Completion blocks should contain the `error` instead of success/failure blocks.
 
 ## Theming
 
-Use a `Theme.swift` that contains extensions for theming your app. When adding color extensions avoid adding the `Color` postfix to your color methods.
+Use a `Theme.swift` that contains extensions for theming your app. When adding color extensions avoid adding the `Color` postfix to your color methods, also add some context of how this color is used, just be careful to not have one method per UI element.
```
Comment (elland) [inline BEST_PRACTICES.md]: Grammar: `just be careful to not have one method per UI element.` should read `just be careful not to have one method per UI element.`.

Comment (3lvis) [inline BEST_PRACTICES.md]: Good catch 💯

**Comment by elland** [comment]
🎉

**Pull request merged**